code/android-core-library/src/main/java/com/adobe/marketing/mobile/internal/configuration/ConfigurationExtension.kt

```diff
@@ -0,0 +1,575 @@
+     * the current app configuration with the resulting downloaded configuration.
+     *
+     * @param event the event requesting/triggering an update to configuration with appId.
+     */
+    private fun configureWithAppID(event: Event) {
+        val appId =
+            event.eventData?.get(Configuration.CONFIGURATION_REQUEST_CONTENT_JSON_APP_ID) as? String
+
+        if (appId.isNullOrBlank()) {
+            MobileCore.log(
+                LoggingMode.VERBOSE,
+                TAG,
+                "AppId in configureWithAppID event is null.."
+            )
+            appIdManager.removeAppIdFromPersistence()
+
+            publishConfigurationState(
+                configurationStateManager.environmentAwareConfiguration,
+                event
+            )
+            return
+        }
+
+        if (!configurationStateManager.hasConfigExpired(appId)) {
+            publishConfigurationState(
+                configurationStateManager.environmentAwareConfiguration,
+                event
+            )
+            return
+        }
+
+        // Stop all event processing for the extension until new configuration download is attempted
+        api.stopEvents()
+
+        configurationStateManager.updateConfigWithAppId(appId) { config ->
+            if (config != null) {
+                cancelConfigRetry()
+                applyConfigurationChanges(event, RulesSource.REMOTE)
+            } else {
+                MobileCore.log(
+                    LoggingMode.VERBOSE,
+                    TAG,
+                    "Failed to download configuration. Applying Will retry download."
+                )
+                // If the configuration download fails, publish current configuration and retry download again.
+                publishConfigurationState(
+                    configurationStateManager.environmentAwareConfiguration,
+                    event
+                )
+                retryConfigTaskHandle = retryConfigDownload(appId)
+            }
+
+            // Start event processing again
+            api.startEvents()
+        }
+    }
+
+    /**
+     * Retrieves the configuration from the file path specified in the [event]'s data and replaces
+     * the current configuration with it.
+     *
+     * @param event the trigger event (whose event data contains the file path for retrieving the config)
+     *              requesting a configuration change
+     */
+    private fun configureWithFilePath(event: Event) {
+        val filePath =
+            event.eventData?.get(Configuration.CONFIGURATION_REQUEST_CONTENT_JSON_FILE_PATH) as String?
+
+        if (filePath.isNullOrBlank()) {
+            MobileCore.log(
+                LoggingMode.WARNING,
+                TAG,
+                "Unable to read config from provided file (filePath is invalid)"
+            )
+            return
+        }
+
+        val config = configurationStateManager.getConfigFromFile(filePath)
```

Comment:
Can you design this api similar to `updateConfigWithAppId` and keep replaceConfigration internal to ConfigurationStateManager